sdk/storage/azblob/blob/client_test.go

```diff
@@ -3012,6 +3012,86 @@ func (s *BlobRecordedTestsSuite) TestBlobSetMetadataIfNoneMatchFalse() {
 	testcommon.ValidateBlobErrorCode(_require, err, bloberror.ConditionNotMet)
 }
 
+func (s *BlobRecordedTestsSuite) TestSetImmutabilityPolicy() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	containerName := testcommon.GenerateContainerName(testName)
+	containerClient := testcommon.GetContainerClient(containerName, svcClient)
+
+	blockBlobName := testcommon.GenerateBlobName(testName)
+	bbClient := testcommon.CreateNewBlockBlob(context.Background(), _require, blockBlobName, containerClient)
+
+	_, err = bbClient.GetProperties(context.Background(), nil)
+	_require.Nil(err)
+
+	currentTime, err := time.Parse(time.UnixDate, "Tue Oct 18 10:00:00 GMT 2022")
+	policy := blob.ImmutabilityPolicySetting(blob.ImmutabilityPolicySettingLocked)
+	_require.Nil(err)
+
+	setImmutabilityPolicyOptions := blob.SetImmutabilityPolicyOptions{
+		ExpiryTime:               &currentTime,
+		Mode:                     &policy,
+		ModifiedAccessConditions: nil,
+	}
+	_, err = bbClient.SetImmutabilityPolicy(context.Background(), &setImmutabilityPolicyOptions)
+	_require.Nil(err)
```

Comment:
It would be good to test if the ImmutabilityPolicy works in the test by attempting to delete the blob (which should fail).